python/lsst/summit/utils/efdUtils.py

```diff
@@ -0,0 +1,566 @@
+        The time at which to get the state.
+    warnStaleAfterNMinutes : `float`, optional
+        The number of minutes after which to consider the data stale and issue
+        a warning.
+
+    Returns
+    -------
+    state : `Enum`
+        The appropriate enumClass for the topic, holding the value it was set
+        to at ``commandTime``.
+    commandTime : `astropy.Time`
+        The time the component was set to the returned state.
+
+    Raises
+    ------
+    ValueError: If the topic is not in the EFD schema.
+    """
+    row = getMostRecentRowWithDataBefore(client=client,
+                                         topic=topic,
+                                         timeToLookBefore=time,
+                                         warnStaleAfterNMinutes=warnStaleAfterNMinutes)
+
+    commandTime = efdTimestampToAstropy(row['private_sndStamp'])
+    # TODO: need to:
+    # a) know which column to get the value from depending on topic, and
+    # b) cast it to the appropriate enum class, and
+    # c) be able to import all the enums, rather than having a few of them them
+    # copied & pasted
+
+    # example: for lsst.sal.MTMount.logevent_azimuthMotionState
+    value = row['state']
+    # state = AxisMotionState(value)
+
+    return value, commandTime
+
+
+def makeEfdClient():
+    """Automatically create an EFD client based on the site.
+
+    Returns
+    -------
+    efdClient : `lsst_efd_client.efd_helper.EfdClient`
+        The EFD client to use for the current site.
+    """
+    if not HAS_EFD_CLIENT:
+        raise RuntimeError("Could not create EFD client because importing lsst_efd_client failed.")
+
+    try:
+        site = getSite()
+    except ValueError as e:
+        raise RuntimeError("Could not create EFD client as the site could not be determined") from e
+
+    if site == 'summit':
+        return EfdClient('summit_efd')
+    if site == 'base':
+        return EfdClient('summit_efd_copy')
+    if site in ['staff-rsp', 'rubin-devl']:
+        return EfdClient('usdf_efd')
+
+    raise RuntimeError(f"Could not create EFD client as the site {site} is not recognized")
+
+
+def expRecordToTimespan(expRecord):
+    """Get the timespan from an exposure record.
+
+    Returns the timespan in a format where it can be used to directly unpack
+    into a efdClient.select_time_series() call.
+
+    Parameters
+    ----------
+    expRecord : `lsst.daf.butler.dimensions.ExposureRecord`
+        The exposure record.
+
+    Returns
+    -------
+    timespanDict : `dict`
+        The timespan in a format that can be used to directly unpack into a
+        efdClient.select_time_series() call.
+    """
+    return {'begin': expRecord.timespan.begin.utc,
+            'end': expRecord.timespan.end.utc,
+            }
+
+
+def efdTimestampToAstropy(timestamp):
+    """Get an efd timestamp as an astropy.time.Time object.
+
+    Parameters
+    ----------
+    timestamp : `float`
+        The timestamp, as a float.
```

Comment:
Since the review, because of comments elsewhere, I've now changed every instance of `private_sndStamp` in the package to use `private_efdStamp`, so everything is now in UTC. I've updated the doc as you sugegst though, but explicitly saying UTC instead of TAI.

Reply:
Ouch. Private_efdStamp may go away Kafka salobj can’t easily support it). Please reconsider

Reply:
Oh wow, I had no idea, thanks for reporting that. I've asked Angelo what he thinks I should do here... I'm off on vacation in an hour or so until Monday, but thank you so much for following up here.